### PR TITLE
BREAKING BETA(LayoutManager): rm `shouldResetTransform`

### PR DESCRIPTION
**Pull request by ShaMan123**
## Description

Thanks to #9561 it is now possible to remove the badly designed `shouldResetTransform` hook from the layout manager, me to blame.
Thanks for the patience!

### CODE REVIEW

File: src/shapes/ActiveSelection.ts [hunk 1]
```diff
@@ -146,12 +147,11 @@ export class ActiveSelection extends Group {
...
 
   /**
-   * If returns true, deselection is cancelled.
-   * @since 2.0.0
-   * @return {Boolean} [cancel]
+   * @override remove all objects and reset transform
    */
   onDeselect() {
     this.removeAll();
+    resetObjectTransform(this);
```
Comment (ShaMan123) [inline src/shapes/ActiveSelection.ts:154]: do we want this?

**Review by ShaMan123 — COMMENTED** [review]
cleanup is done
now decisions

**Comment by ShaMan123** [comment]
I have noticed that we introduced a regression with polys flagged with exactBoundingBox

**Review by ShaMan123 — COMMENTED** [review]
removed resetting transform, ready to go

**Comment by ShaMan123** [comment]
We can bind, sure. You want it in this PR?

> My understanding is that it takes care of the the size, of the group and the position of the group.

position of the children, size of the group so doing stuff like table or flex layout is part of the scope.
We (@ebigtm and I) did a flex one already and we might publish it.
A table layout is part of what a project I am involved with in doing now.
I also saw @iocoker do a calendar, which is a table layout.

So it is possible and the API is pretty solid so hooking into should be relatively straight forward.
The only pain is handling the correction, that makes it just hard for "no reason".
That is why I am thinking, what if we use left top origin instead of center. Maybe that will make it easier.
And I will try this small hack I have in mind that is equiv to `pathOffset` that might release us from the need to correct the position

File: src/shapes/ActiveSelection.ts [hunk 2]
```diff
@@ -146,9 +146,7 @@ export class ActiveSelection extends Group {
   }
 
   /**
-   * If returns true, deselection is cancelled.
-   * @since 2.0.0
-   * @return {Boolean} [cancel]
+   * @override remove all objects
```
Comment (asturur) [inline src/shapes/ActiveSelection.ts:149]: why removing the comment?

**Comment by asturur** [comment]
We can make another pr for the binding issue

File: src/shapes/ActiveSelection.ts (hunk 2 above)
Comment (ShaMan123) [reply src/shapes/ActiveSelection.ts:149]: I think the jsdoc concats the comments of the super method so there is no need, it is spam

**Comment by ShaMan123** [comment]
> > I have noticed that we introduced a regression with polys flagged with exactBoundingBox
> 
> What is the regression?

I should open a ticket.

1. stroke uniform poly
2. perform multiselection, select and deselect the poly
3. poly with jump a few pixels, maybe related to the stroke width value

**Comment by ShaMan123** [comment]
So can we merge?

File: src/shapes/ActiveSelection.ts (hunk 2 above)
Comment (asturur) [reply src/shapes/ActiveSelection.ts:149]: that is bad, because if i m just reading the codebase i can't concat anything

File: src/LayoutManager/LayoutManager.spec.ts [hunk 3]
```diff
@@ -305,19 +305,27 @@ describe('Layout Manager', () => {
...
 
-      const target = new Group([], { scaleX: 2, scaleY: 0.5, angle: 30 });
+      const rect = new FabricObject({ width: 50, height: 50 });
+      const target = new Group([rect], { scaleX: 2, scaleY: 0.5, angle: 30 });
 
       const context: StrictLayoutContext = {
         bubbles: true,
         strategy: manager.strategy,
         target,
+        targets: [rect],
```
Comment (asturur) [inline src/LayoutManager/LayoutManager.spec.ts:315]: The layoutManager.getLayoutResult is not using targets at all, it is using target.getObjects().

Comment (ShaMan123) [reply src/LayoutManager/LayoutManager.spec.ts:315]: hmmm
that is maybe something to discuss
targets are objects that changed (I think it is used only for subscribing)

File: src/shapes/ActiveSelection.ts (hunk 2 above)
Comment (ShaMan123) [reply src/shapes/ActiveSelection.ts:149]: And if you hover with your mouse?

Comment (ShaMan123) [reply src/shapes/ActiveSelection.ts:149]: Screenshot 2024-01-11 at 10 32 50
 I can restore it but my point is that then you have to update a signle comment everywhere 
That is not manageable

File: src/LayoutManager/LayoutManager.spec.ts (hunk 3 above)
Comment (asturur) [reply src/LayoutManager/LayoutManager.spec.ts:315]: Adding or removing this line doesn't do anything else than helping TS, is also a bit confusing in the test definition that one test 'LAYOUT_TYPE_INITIALIZATION ' pass empty targets to override this one here while 'LAYOUT_TYPE_IMPERATIVE' doesn't.

Also on my vscode both '[]' and '[rect]' give a red error in typescript.

I will investigate this typing issue right after merging, i don't think a function that doesn't use 'targets' should requires it in the typings

File: src/shapes/ActiveSelection.ts (hunk 2 above)
Comment (asturur) [reply src/shapes/ActiveSelection.ts:149]: yeah VSCODE does this, i m just not sure how it is supposed to be by the JSDOCS logic

**Review by asturur — APPROVED** [review]
